.github/workflows/tracegen.yml

```diff
@@ -0,0 +1,60 @@
+name: tracegen
+on:
+  push:
+    branches: [ main ]
+    tags:
+      - 'v[0-9]+.[0-9]+.[0-9]+*'
+  pull_request:
+
+jobs:
+  build-dev:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout Repo
+        uses: actions/checkout@v2
+      - name: Build tracegen
+        run: |
+          docker build tracegen -t ghcr.io/open-telemetry/opentelemetry-collector-contrib/tracegen:dev
+
+  publish-dev:
+    runs-on: ubuntu-latest
+    if: (github.ref == 'refs/heads/main' || startsWith(github.ref, 'refs/tags/v')) && github.repository == 'open-telemetry/opentelemetry-collector-contrib'
```

Comment:
You sure we need this? I think the metadata action should handle this for us: https://github.com/docker/metadata-action#usage

Reply:
We still use `if` on many occasions: https://github.com/open-telemetry/opentelemetry-collector-contrib/blob/main/.github/workflows/build-and-test.yml#L356, I'd like to use this to keep this consistent with others. And this action is mostly for tagging and labeling docker images.